src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -755,6 +777,9 @@ def __call__(
         text_encoder_lora_scale = (
             cross_attention_kwargs.get("scale", None) if cross_attention_kwargs is not None else None
         )
+        # 3.1 Dynamically adjust the LoRA scale
+        self._adjust_lora_scale_text_encoder(self.text_encoder, text_encoder_lora_scale)
+        self._adjust_lora_scale_text_encoder(self.text_encoder_2, text_encoder_lora_scale)
```

Comment:
This is the only easy and simple way I could think of. Happy to port this changes to over to the other pipelines if this works.

Reply:
That works for me, but could we maybe move this into the `encode_prompt` function? This way:
- a) Copied from will take care for most of the changes for us
- b) It's cleaner IMO because it's only needed when `encode_prompt` is called